irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -42,9 +53,18 @@ namespace iroha {
                 }
                 |
                 [&](const auto &signatories) {
+                  auto shm_signatories =
```

Comment:
Avoid using shared model in variable names, otherwise can have some inconsistency if [we decided renaming](https://github.com/hyperledger/iroha/issues/967#issuecomment-365215540)